services/management/node.go

```diff
@@ -63,6 +63,17 @@ func (s *NodeService) Register(ctx context.Context, req *managementpb.RegisterNo
 			return err
 		}
 
+		if node, err := models.CheckUniqueNodeInstanceRegion(tx.Querier, req.Address, req.Region); err != nil {
+			if status.Code(err) == codes.AlreadyExists && req.Reregister {
+				if err := models.RemoveNode(tx.Querier, node.NodeID, models.RemoveCascade); err != nil {
```

Comment:
`err` is shadowed there. Typically, it is better avoid that. Seems like `err =` will work there.